Bug in 'Developing a “TODO” extension' tutorial

**Describe the bug**
I followed the tutorial. All was well until I changed the line in `conf.py` from

`todo_include_todos = False`

to

`todo_include_todos = True`

**To Reproduce**
Steps to reproduce the behavior:
```

Follow the tutorial, then type:

$ make html
```

This causes an Exception:

```
➜  sphinx make html
Running Sphinx v2.4.4
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [100%] index                                                                                        
Exception occurred:
  File "/Users/matthewgillman/play/sphinx/_ext/todo.py", line 82, in process_todo_nodes
    for todo_info in env.todo_all_todos:
AttributeError: 'BuildEnvironment' object has no attribute 'todo_all_todos'
The full traceback has been saved in /var/folders/kj/7h5hgyvn4v9bjtyk88y4s1rh0000gn/T/sphinx-err-answiaoe.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2
```

[sphinx-err-answiaoe.log](https://github.com/sphinx-doc/sphinx/files/4319346/sphinx-err-answiaoe.log)


**Expected behavior**
The tutorial should work

**Environment info**
- OS: Mac Darwin
- Python version: 3.6.10 |Anaconda, Inc.| (default, Jan  7 2020, 15:01:53) 
- Sphinx version: sphinx-build 2.4.4
- Sphinx extensions:  extensions = ['todo']

**Additional context**
See attached error log

--- a/doc/development/tutorials/examples/todo.py
+++ b/doc/development/tutorials/examples/todo.py
@@ -70,6 +70,9 @@ def process_todo_nodes(app, doctree, fromdocname):
     # Augment each todo with a backlink to the original location.
     env = app.builder.env
 
+    if not hasattr(env, 'todo_all_todos'):
+        env.todo_all_todos = []
+
     for node in doctree.traverse(todolist):
         if not app.config.todo_include_todos:
             node.replace_self([])

